models/daily_spend.sql

```diff
@@ -282,16 +456,28 @@ search_optimization_spend_daily as (
         null as storage_type,
         null as warehouse_name,
         null as database_name,
-        coalesce(sum(stg_metering_history.credits_used * daily_rates.effective_rate), 0) as spend,
+        coalesce(
+            sum(
+                stg_metering_history.credits_used * coalesce(
+                    daily_rates.effective_rate, latest_rates.effective_rate
+                )
+            ),
+            0
+        ) as spend,
         spend as spend_net_cloud_services
     from dates
     left join {{ ref('stg_metering_history') }} on
-        dates.date = convert_timezone('UTC', stg_metering_history.start_time)::date
+        dates.date = convert_timezone(
+            'UTC', stg_metering_history.start_time
+        )::date
         and stg_metering_history.service_type = 'SEARCH_OPTIMIZATION'
-    inner join {{ ref('daily_rates') }}
+    left join {{ ref('daily_rates') }}
         on dates.date = daily_rates.date
             and daily_rates.service_type = 'COMPUTE'
-            and daily_rates.usage_type = 'search optimization`'
+            and daily_rates.usage_type = 'search optimization'
+    inner join latest_rates
+        on latest_rates.service_type = 'COMPUTE'
+            and latest_rates.usage_type = 'search optimization'
     group by 1, 2, 3, 4
 )
 
```

Comment:
@NiallRees fyi sqlfluff was failing for this commit, couldn't tell why since all the violations it was complaining about existed before. Let's pair when you're back, want to get this out since the issue has been holding up some dashboards from updating.

```
SQLFluff Fix.............................................................Failed
- hook id: sqlfluff-fix
- exit code: 1

fix installed: agate==1.6.3,appdirs==1.4.4,asn1crypto==1.5.1,attrs==22.1.0,Babel==2.11.0,certifi==2022.12.7,cffi==1.15.1,chardet==5.1.0,charset-normalizer==2.1.1,click==8.1.3,colorama==0.4.5,cryptography==36.0.2,dbt-core==1.3.1,dbt-extractor==0.4.1,dbt-snowflake==1.3.0,diff-cover==7.3.0,exceptiongroup==1.0.4,filelock==3.8.2,future==0.18.2,hologram==0.0.15,idna==3.4,importlib-metadata==5.1.0,iniconfig==1.1.1,isodate==0.6.1,jaraco.classes==3.2.3,Jinja2==3.1.2,jinja2-simple-tags==0.4.1,jsonschema==3.2.0,keyring==23.11.0,leather==0.3.4,Logbook==1.5.3,MarkupSafe==2.1.1,mashumaro==3.0.4,minimal-snowplow-tracker==0.0.2,more-itertools==9.0.0,msgpack==1.0.4,networkx==2.8.8,oscrypto==1.3.0,packaging==21.3,parsedatetime==2.4,pathspec==0.9.0,pluggy==1.0.0,pycparser==2.21,pycryptodomex==3.16.0,Pygments==2.13.0,PyJWT==2.6.0,pyOpenSSL==22.0.0,pyparsing==3.0.9,pyrsistent==0.19.2,pytest==7.2.0,python-dateutil==2.8.2,python-slugify==7.0.0,pytimeparse==1.1.8,pytz==2022.6,PyYAML==6.0,regex==2022.10.31,requests==2.28.1,six==1.16.0,snowflake-connector-python==2.7.12,sqlfluff==1.4.4,sqlfluff-templater-dbt==1.4.4,sqlparse==0.4.3,tblib==1.7.0,text-unidecode==1.3,toml==0.10.2,tomli==2.0.1,tqdm==4.64.1,typing_extensions==4.4.0,urllib3==1.26.13,Werkzeug==2.2.2,zipp==3.11.0
fix run-test-pre: PYTHONHASHSEED='564625851'
fix run-test: commands[0] | sqlfluff fix models/daily_spend.sql --ignore parsing -f
==== finding fixable violations ====
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Tried to denylist unknown rules: ['LT05', 'ST06', 'RF04', 'AM06', 'ST05', 'LT02'] 
==== no fixable linting violations found ==== 
All Finished 📜 🎉!
  [33 unfixable linting violations found]
ERROR: InvocationError for command /Users/ianwhitestone/git/get-select/dbt-snowflake-monitoring/.tox/fix/bin/sqlfluff fix models/daily_spend.sql --ignore parsing -f (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   fix: commands failed


SQLFluff Lint............................................................Failed
- hook id: sqlfluff-lint
- exit code: 1

lint installed: agate==1.6.3,appdirs==1.4.4,asn1crypto==1.5.1,attrs==22.1.0,Babel==2.11.0,certifi==2022.12.7,cffi==1.15.1,chardet==5.1.0,charset-normalizer==2.1.1,click==8.1.3,colorama==0.4.5,cryptography==36.0.2,dbt-core==1.3.1,dbt-extractor==0.4.1,dbt-snowflake==1.3.0,diff-cover==7.3.0,exceptiongroup==1.0.4,filelock==3.8.2,future==0.18.2,hologram==0.0.15,idna==3.4,importlib-metadata==5.1.0,iniconfig==1.1.1,isodate==0.6.1,jaraco.classes==3.2.3,Jinja2==3.1.2,jinja2-simple-tags==0.4.1,jsonschema==3.2.0,keyring==23.11.0,leather==0.3.4,Logbook==1.5.3,MarkupSafe==2.1.1,mashumaro==3.0.4,minimal-snowplow-tracker==0.0.2,more-itertools==9.0.0,msgpack==1.0.4,networkx==2.8.8,oscrypto==1.3.0,packaging==21.3,parsedatetime==2.4,pathspec==0.9.0,pluggy==1.0.0,pycparser==2.21,pycryptodomex==3.16.0,Pygments==2.13.0,PyJWT==2.6.0,pyOpenSSL==22.0.0,pyparsing==3.0.9,pyrsistent==0.19.2,pytest==7.2.0,python-dateutil==2.8.2,python-slugify==7.0.0,pytimeparse==1.1.8,pytz==2022.6,PyYAML==6.0,regex==2022.10.31,requests==2.28.1,six==1.16.0,snowflake-connector-python==2.7.12,sqlfluff==1.4.4,sqlfluff-templater-dbt==1.4.4,sqlparse==0.4.3,tblib==1.7.0,text-unidecode==1.3,toml==0.10.2,tomli==2.0.1,tqdm==4.64.1,typing_extensions==4.4.0,urllib3==1.26.13,Werkzeug==2.2.2,zipp==3.11.0
lint run-test-pre: PYTHONHASHSEED='270690111'
lint run-test: commands[0] | sqlfluff lint models/daily_spend.sql --ignore parsing
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Tried to denylist unknown rules: ['LT05', 'ST06', 'RF04', 'AM06', 'ST05', 'LT02'] 
== [models/daily_spend.sql] FAIL              
L:  61 | P:  21 | L029 | Keywords should not be used as identifiers.
L:  81 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L:  89 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L:  97 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 101 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 129 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 133 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 161 | P: 120 | L016 | Line is too long.
L: 162 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 166 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 193 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 197 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 205 | P:  89 | L016 | Line is too long.
L: 222 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 226 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 232 | P:  93 | L016 | Line is too long.
L: 255 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 268 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 278 | P: 119 | L016 | Line is too long.
L: 285 | P:  83 | L016 | Line is too long.
L: 293 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 321 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 325 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 349 | P: 140 | L016 | Line is too long.
L: 353 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 357 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 385 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 389 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 417 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 421 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 449 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
L: 453 | P:   5 | L034 | Select wildcards then simple targets before calculations
                       | and aggregates.
L: 481 | P:   5 | L054 | Inconsistent column references in 'GROUP BY/ORDER BY'
                       | clauses.
All Finished 📜 🎉!
ERROR: InvocationError for command /Users/ianwhitestone/git/get-select/dbt-snowflake-monitoring/.tox/lint/bin/sqlfluff lint models/daily_spend.sql --ignore parsing (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   lint: commands failed
```